include/bluetooth/audio/tbs.h

```diff
@@ -328,4 +328,411 @@ void bt_tbs_register_cb(struct bt_tbs_cb *cbs);
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_signal_strength(struct bt_conn *conn,
+				       uint8_t inst_index);
+
+/**
+ * @brief Read the signal strength reporting interval of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_signal_interval(struct bt_conn *conn,
+				       uint8_t inst_index);
+
+/**
+ * @brief Read the list of current calls of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_current_calls(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the content ID of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_ccid(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the feature and status value of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_status_flags(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the call target URI of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_call_uri(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the states of the current calls of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_call_state(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the remote URI of a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_remote_uri(struct bt_conn *conn, uint8_t inst_index);
+
+/**
+ * @brief Read the friendly name of a call for a TBS instance.
+ *
+ * @param conn          The connection to the TBS server.
+ * @param inst_index    The index of the TBS instance.
+ *
+ * @return              int 0 on success, errno value on fail.
+ */
+int bt_tbs_client_read_friendly_name(struct bt_conn *conn, uint8_t inst_index);
+
+/** @brief Read the supported opcode of a TBS instance.
```

Comment:
Shouldn't this be handled internally, and not exposed by API? My feeling is that if client supports (see my other comment related to client configurable features) any of the *optional* CP operations the characteristic could be read internally. So that if API user will try to e.g. Local Hold the implementation would respond with an error immediately (not even writing to the CP).

Reply:
That's not a bad idea. The API was implemented to be as flexible as possible, and not do anything unnecessary and following the spec defined procedures. e.g. if you are doing a client that only uses the mandatory features, you wouldn't need to do this at all. 

Doing it autonomously would be somewhat conflicting with your other comment https://github.com/zephyrproject-rtos/zephyr/pull/40707#discussion_r764671350

Reply:
It would not conflict. It could be done autonomously if the Client would compile the code with Hold and/or Join features enabled.

Reply:
Good point :) It's definitely something we can change in the future. This does not need to be an API indeed.

Reply:
how do we track changes we want to do in the future?

Reply:
@rymanluk Excellent question. Let's discuss that in tomorrow meeting. We are tracking it internally in Nordic, but we may need to double track it somehow.